uv: enable build on z/OS

zsw007 opened the pull request:
Initial changes to enable libuv to build on z/OS. The GYP changes allows libuv to be built as a part of Node.js. The CMake changes enable libuv to be built standalone, and makes it possible to run the unit tests. Since libuv requires the `csrsic.h` file, which is provided by zoslib, the option `-DZOSLIB_DIR` was added to CMake to allow the user to specify the path to zoslib when building standalone.

zsw007 committed:
uv,zos: use correct file status timestamp fields

On z/OS, the older standard of st_atime, st_mtime, and st_ctime are used
instead of st_atim, st_mtim, and st_ctim.

zsw007 committed:
uv,zos: add gyp cflags configuration for z/OS

zsw007 committed:
uv,zos: fix CMake configuration for z/OS

IgorTodorovskiIBM commented:
Looks good, should we place csrsic.h in zoslib since it has an IBM copyright and is zOS only?  https://www.ibm.com/support/knowledgecenter/SSLTBW_2.1.0/com.ibm.zos.v2r1.ieaa100/CSRSIC_C_370_header_file.htm

alexcfyung commented:
LGTM

zsw007 committed:
uv,zos: allow headers to be included from zoslib

The required header file `csrsic.h` can be found in zoslib. Allow users
to specify the path to zoslib using `-DZOSLIB_DIR` flag so the header
can be included.

zsw007 commented:
> Looks good, should we place csrsic.h in zoslib since it has an IBM copyright and is zOS only? https://www.ibm.com/support/knowledgecenter/SSLTBW_2.1.0/com.ibm.zos.v2r1.ieaa100/CSRSIC_C_370_header_file.htm

Makes sense to me. Added the option `-DZOSLIB_DIR` to CMake to allow the user to specify the zoslib path when building standalone. 

Pending https://github.com/ibmruntimes/zoslib/pull/2

IgorTodorovskiIBM reviewed (approved):
LGTM

gabylb reviewed (approved):
LGTM
